FIX: Remove AutoShun from IntelMQ as the feed is discontinued

waldbauer-certat opened the pull request:
Feed is discontinued, removing Parser, adding upgrade information.

Fixes #2162

waldbauer-certat committed:
FIX: Remove AutoShun from IntelMQ as the feed is discontinued

Fixes #2162

Signed-off-by: Sebastian Waldbauer <waldbauer@cert.at>

sebix commented on intelmq/lib/upgrades.py:
```diff
@@ -725,6 +726,29 @@ def v301_deprecations(configuration, harmonization, dry_run, **kwargs):
+        if bot_id == 'global':
+            continue
+        if bot["module"] == "intelmq.bots.collectors.http.collector":
+            if "http_url" not in bot["parameters"]:
+                continue
+            if 'autoshun.org' in bot["parameters"]["http_url"]:
```
```suggestion
            if bot["parameters"].get("http_url", "").startswith("https://www.autoshun.org/download"):
```

sebix commented on CHANGELOG.md:
```diff
@@ -72,6 +72,7 @@ CHANGELOG
 - `intelmq.bots.parsers.shadowserver._config`:
   - Added support for `Accessible AMQP`, `Device Identification Report` (IPv4 and IPv6) (PR#2134 by Mateo Durante).
   - Added file name mapping for `SSL-POODLE-Vulnerable-Servers IPv6` (file name `scan6_ssl_poodle`) (PR#2134 by Mateo Durante).
+- `intelmq.bots.parsers.autoshun.parser`: Removed, as the feed is discontinued (PR#2214 by Sebastian Waldbauer).
```
```suggestion
- `intelmq.bots.parsers.autoshun.parser`: Removed, as the feed is discontinued (PR#2214 by Sebastian Waldbauer, fixes#2162).
```

sebix reviewed (changes requested):
Missing tests for the upgrade function

sebix commented:
Rebase on develop to fix conflict

sebix committed:
tst codespell: update excludelines

intelmq/tests/lib/test_upgrades.py was reformatted by @waldbauer-certat
in 1eb14d6f7195d319237f23652dcabdacd29b9033, fix the excludelines for
codespell now

sebix commented:
One unittest is failing with

> Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v2/dist/codecov' failed with exit code 255

We don't care.
